### PR TITLE
debian,vendor: import github.com/snapcore/squashfs and use

### PR DESCRIPTION
**Pull request by mvo5**
This PR will add a new vendor dependency github.com/snapcore/squashfs
that is used to provide squashfuse for lxd.

It superseeds https://github.com/snapcore/snapd/pull/4030 and still needs some love but for early review it should be ok. Some trickery needed to make github.com/snapcore/squashfuse `go get` able (https://github.com/snapcore/squashfuse/commit/ed6f37bbee6537e0598aa7f4f5dafa379be0c73a).

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@mvo5 This is a bit less gross, but why do this through the go vendor?

**Comment by pedronis** [comment]
marked as blocked as apparently it's not fully working and we need to look into a different approach according to @mvo5

**Comment by mvo5** [comment]
This needs a forum topic with "upcoming" tag and then can be closed.

**Comment by zyga** [comment]
@mvo5 shall we close this or do you think it is a viable approach but just needs more work?

**Comment by Conan-Kudo** [comment]
Please don't do it this way. If you must vendor it, do it the more standard way (add an additional tarball to the dsc packaging and build it appropriately).

Shoehorning into weird Go things is just horrible.

**Comment by mvo5** [comment]
@Conan-Kudo Yes, sorry for this. Unfortunately there is no clean deb package way to vendor this, this hack is about the cleanest solution I was able to find :/

**Comment by Conan-Kudo** [comment]
@mvo5 Last I checked, you should be able to just attach more source tarballs in the dsc description when uploading to Ubuntu. The rules file can be adjusted to actually use the second source vendor it in. In addition, this change should be considered temporary and dropped for bionic, as that release can just use regular squashfuse. Same goes for Debian sid.

There's no good reason to make a permanent hack here.

File: packaging/ubuntu-16.04/control [hunk 1]
```diff
@@ -20,9 +20,11 @@ Build-Depends: autoconf,
                golang-any (>=2:1.6) | golang-1.6,
                indent,
                init-system-helpers,
-               libcap-dev,
                libapparmor-dev,
+               libcap-dev,
+               libfuse-dev,
```
Comment (zyga) [inline packaging/ubuntu-16.04/control]: I see this is in 16.04 build, what about 14.04? I know it doesn't build but do you anticipate any problems due to its absence?

**Comment by zyga** [comment]
@mvo5 is this suitable for 2.31 or too risky?

**Pull request merged**